test/test_generated_mypy.py

```diff
@@ -310,6 +320,17 @@ def test_which_oneof_proto3():
         s_untyped.WhichOneof("garbage")
 
 
+def test_extensions_proto2():
+    # type: () -> None
+    s = Simple1()
+
+    e1 = s.Extensions[Extensions1.ext]  # type: Extensions1
+    e1.ext1_string = "first extension"
+
+    e2 = s.Extensions[Extensions2.foo]  # Will be type cast as an Any.
```

Comment:
If `Extensions` is typed, then e2 can by typed as a Message here.
It'll probably require some manual casting by the caller to get the appropriate message type, but I don't really see a way around that. I think it's probably better to force the caller to do an explicit type cast from Message to the appropriate Message type, over silently returning Any here.

In any case, what you have so far is better than what we had before, so I'm ok to run with it.

Reply:
I figured out a way around it using generics :).  Now `e2` is inferred to be an `Extensions2`.